week2/homework/bonus/bonus.js

```diff
@@ -0,0 +1,77 @@
+const program = require('commander');
+const fs = require('fs');
+
+program
+  .option('add <new todo>', 'Add new todo item', function(todoItem) {
+    fs.readFile('./bonus/list.json', 'utf8', (error, toDos) => {
+      if (error) {
+        console.log(error);
+      }
+      const todoList = JSON.parse(toDos);
+      todoList.push(todoItem);
+      const newList = JSON.stringify(todoList, null, 2);
+      console.log(newList);
+      return fs.writeFile('./bonus/list.json', newList, error => console.log(error));
+    });
+  })
+  .option('list', 'List all todo items', function() {
+    fs.readFile('./bonus/list.json', 'utf8', (error, toDos) => {
+      if (error) {
+        console.log(error);
+      }
+      console.log(toDos);
+    });
+  })
+  .option('reset', 'Removes all todo items', function() {
+    fs.writeFile('./bonus/list.json', '[]', error => console.log(error));
+  })
+  .option('remove <todo row>', 'Remove one todo item', function(todoItem) {
+    fs.readFile('./bonus/list.json', 'utf8', (error, toDos) => {
+      if (error) {
+        console.log(error);
+      }
+      const todoList = JSON.parse(toDos);
+      const filteredList = todoList.filter((elem, index) => index !== Number(todoItem) - 2);
+      const newList = JSON.stringify(filteredList, null, 2);
+      console.log(newList);
+      return fs.writeFile('./bonus/list.json', newList, error => console.log(error));
+    });
+  })
+  // The `update` takes only one argument, the second one is `undefined`.
+  // When I try to update an item it becomes `null`
+  .option('update <row number> <new value>', 'update one todo item', function(todoItem, newVal) {
+    fs.readFile('./bonus/list.json', 'utf8', (error, toDos) => {
+      if (error) {
+        console.log(error);
+      }
+      const todoList = JSON.parse(toDos);
+      const mappedList = todoList.map((elem, index) =>
+        index === Number(todoItem) - 2 ? newVal : elem,
```

Comment:
Why - 2 ?

Reply:
To reach `todoItem` by its line number. The first line of the `json` file is `[` and the first `todoItem` is line `2`

Reply:
But you are using the index for the json array not for the line number in the file

Reply:
So `-1` will be enough this way.
I'll do that.